src/platform/packages/private/kbn-language-documentation/scripts/generate_esql_command_docs.ts

```diff
@@ -0,0 +1,252 @@
+ */
+function addDefinitionsToCommands(
+  data: CommandsData,
+  definitions: Map<string, CommandDefinition>
+): CommandsData {
+  return {
+    ...data,
+    items: data.items.map((item) => {
+      const commandDef = definitions.get(item.name);
+      const licenseInfo = getLicenseInfoForCommand(commandDef);
+
+      if (licenseInfo) {
+        return {
+          ...item,
+          license: licenseInfo,
+        };
+      }
+      return item;
+    }),
+  };
+}
+
+/**
+ * Generates the full content for a single command documentation file.
+ */
+function generateDoc(data: CommandsData): string {
+  return `
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { Markdown as SharedUXMarkdown } from '@kbn/shared-ux-markdown';
+
+const Markdown = (props: Parameters<typeof SharedUXMarkdown>[0]) => (
+  <SharedUXMarkdown {...props} readOnly enableSoftLineBreaks />
+);
+
+// Do not edit manually... automatically generated by scripts/generate_esql_command_docs.ts
+
+export const commands = ${generateCommandsDoc(data)};
+`;
+}
+
+/**
+ * Generates a documentation for a specific group of commands.
+ */
+function generateCommandsDoc({
+  items,
+  labelKey,
+  labelDefaultMessage,
+  descriptionKey,
+  descriptionDefaultMessage,
+}: CommandsData): string {
+  const commandsContentDoc = items.map((item) => generateCommandDoc(item)).join(',\n    ');
+
+  return `{
+  label: i18n.translate('${labelKey}', {
+    defaultMessage: '${labelDefaultMessage}',
+  }),
+  description: i18n.translate('${descriptionKey}', {
+    defaultMessage: \`${descriptionDefaultMessage}\`,
+  }),
+  items: [
+    ${commandsContentDoc}
+  ],
+}`;
+}
+
+/**
+ * Generates a single command documentation.
+ */
+function generateCommandDoc({
+  name,
+  labelDefaultMessage,
+  descriptionDefaultMessage,
+  openLinksInNewTab,
+  descriptionOptions,
+  preview,
+  license,
+}: CommandData): string {
+  function generateMarkdownProps(props: { openLinksInNewTab?: boolean }): string {
+    return props.openLinksInNewTab ? ' openLinksInNewTab={true}' : '';
+  }
+
+  function generateDescriptionOptions(
+    options: {
+      ignoreTag?: boolean;
+      description?: string;
+    } = {}
+  ) {
+    const formattedDescriptionOptions = Object.entries(options || {}).map(([key, value]) =>
+      typeof value === 'boolean'
+        ? `${key}: ${value},`
+        : `${key}: '${String(value).replace(/'/g, "\\'")}',`
+    );
+
+    return formattedDescriptionOptions.length > 0
+      ? `,\n ${formattedDescriptionOptions.join('\n')}`
+      : '';
+  }
+
+  /**
+   * Generates labelKey and descriptionKey from command name, handling special cases.
+   * These special cases are necessary to ensure that the i18n keys follow the naming conventions from the Elasticsearch definitions.
+   */
+  function generateLabelAndDescriptionKeys(cmdName: string): {
+    labelKey: string;
+    descriptionKey: string;
+  } {
+    const labelKeySuffixMap: Record<string, string> = {
```

Comment:
I'm not a big fan of these bindings, but the only alternative is to change the names of the translation keys in the translation json files

Reply:
We need to handle cases where the label keys in the translation files don’t match the command names exactly. For example, a label key might be "languageDocumentation.documentationESQL.lookupJoin", but the corresponding commandName is simply lookup.

In most cases, the label keys and command names follow a predictable pattern and can be generated programmatically. However, there are exceptions like the one above, where a manual mapping is required to ensure accuracy.

As part of this, I’ll be updating the keys in the translation files to align with these mappings.